assets/js/modules/adsense/setup/setup-main.js

```diff
@@ -0,0 +1,293 @@
+	const alerts = useSelect( ( select ) => select( STORE_NAME ).getAlerts( accountID ) );
+	const urlChannels = useSelect( ( select ) => select( STORE_NAME ).getURLChannels( clientID ) );
+	const error = useSelect( ( select ) => select( STORE_NAME ).getError() );
+
+	// Determine account and site status.
+	const accountStatus = determineAccountStatus( {
+		accounts,
+		clients,
+		alerts,
+		error,
+		previousAccountID,
+		previousClientID,
+	} );
+	const siteStatus = determineSiteStatus( {
+		urlChannels,
+		siteURL,
+	} );
+
+	const {
+		setAccountID,
+		setClientID,
+		setAccountStatus,
+		setSiteStatus,
+		setAccountSetupComplete,
+		setSiteSetupComplete,
+		setUseSnippet,
+		submitChanges,
+	} = useDispatch( STORE_NAME );
+
+	// Update current account ID setting on-the-fly.
+	useEffect( () => {
+		// Don't do anything if setting has not loaded yet or if account ID cannot be determined.
+		if ( 'undefined' === typeof previousAccountID || 'undefined' === typeof accountID ) {
+			return;
+		}
+		setAccountID( accountID );
+	}, [ previousAccountID, accountID ] );
+
+	// Update current client ID setting on-the-fly.
+	useEffect( () => {
+		// Don't do anything if setting has not loaded yet or if client ID cannot be determined.
+		if ( 'undefined' === typeof previousClientID || 'undefined' === typeof clientID ) {
+			return;
+		}
+		setClientID( clientID );
+	}, [ previousClientID, clientID ] );
+
+	// Update account status setting on-the-fly.
+	useEffect( () => {
+		// Don't do anything if account status cannot be determined (because of arguments not loaded yet).
+		if ( 'undefined' === typeof accountStatus ) {
+			return;
+		}
+		// Force setup completion flags to false in case it had been set before.
+		if ( accountStatus !== ACCOUNT_STATUS_APPROVED ) {
+			setAccountSetupComplete( false );
+			setSiteSetupComplete( false );
+		}
+		// Force snippet placement to true when account is graylisted or pending.
+		if ( accountStatus === ACCOUNT_STATUS_GRAYLISTED || accountStatus === ACCOUNT_STATUS_PENDING ) {
+			setUseSnippet( true );
+		}
+		setAccountStatus( accountStatus );
+	}, [ accountStatus ] );
+
+	// Update site status setting on-the-fly.
+	useEffect( () => {
+		// Don't do anything if site status cannot be determined (because of arguments not loaded yet).
+		if ( 'undefined' === typeof siteStatus ) {
+			return;
+		}
+		// Force site setup completion flag to false in case it had been set before.
+		if ( siteStatus !== SITE_STATUS_ADDED ) {
+			setSiteSetupComplete( false );
+		}
+		setSiteStatus( siteStatus );
+	}, [ siteStatus ] );
+
+	// Submit changes for determined parameters in the background when they are valid.
+	const [ isSubmittingInBackground, setIsSubmittingInBackground ] = useState( false );
+	useEffect( () => {
+		// Only submit changes if valid.
+		// TODO: Remove temporary hack to avoid saving in Storybook.
+		if ( ! canSubmitChanges || global.__STORYBOOK_ADDONS ) {
```

Comment:
@aaemnnosttv @tofumatt I wanted to avoid firing requests like this in storybook and added this hack. 😆 Do you have any better idea?

Reply:
We have some kind of fetch mock installed which is usable in Storybook, although we're not currently using it I don't think so you might need to look up how to use it. I forget exactly what it's called, it's named very similar to the one we use in Jest but slightly different 😆 - @tofumatt can you point him to the relevant docs?

Reply:
Mocking fetch _in_ storybook seems the right approach—I think @aaemnnosttv did that already in the Analytics code. But I can't find it now that I'm looking for it. If @aaemnnosttv knows off the top of his head that's cool, otherwise I'll circle back to this later 😄

Reply:
The more generic `fetch-mock` library we have loaded in storybook is: https://www.npmjs.com/package/fetch-mock. API docs are here: http://www.wheresrhys.co.uk/fetch-mock/ and there's a quick start here: https://github.com/wheresrhys/fetch-mock/blob/master/docs/cheatsheet.md

Looks like the very basics are:

```js
fetchMock
  .get('/expected-url', {status: 200, statusText: 'responseText' });
```

http://www.wheresrhys.co.uk/fetch-mock/#api-mockingget_post goes more in-depth. That should allow you to remove all the `global.__STORYBOOK_ADDONS` checks though. If it's not behaving well happy to pair with you or check out the code that isn't working if you want to push a WIP commit 🙂